### PR TITLE
feat: Add multi-schema file support

### PR DESCRIPTION
**Pull request by jiashengguo**

### CODE REVIEW

File: packages/schema/src/language-server/validator/schema-validator.ts [hunk 1]
```diff
@@ -1,16 +1,31 @@
...
     validate(model: Model, accept: ValidationAcceptor): void {
         validateDuplicatedDeclarations(model.declarations, accept);
 
+        const importedModels = resolveTransitiveImports(this.documents, model);
+
+        const importedNames = new Set(importedModels.flatMap((m) => m.declarations.map((d) => d.name)));
+
+        for (const declaration of model.declarations) {
+            if (importedNames.has(declaration.name)) {
+                accept('error', `A ${declaration.name} already exists in an imported models`, {
```
Comment (ymc9) [inline packages/schema/src/language-server/validator/schema-validator.ts:22]: models -> module?

File: packages/schema/src/language-server/validator/schema-validator.ts [hunk 2]
```diff
@@ -20,10 +35,8 @@ export default class SchemaValidator implements AstValidator<Model> {
     }
 
     private validateDataSources(model: Model, accept: ValidationAcceptor) {
-        const dataSources = model.declarations.filter((d) => isDataSource(d));
-        if (dataSources.length === 0) {
-            accept('error', 'Model must define a datasource', { node: model });
-        } else if (dataSources.length > 1) {
```
Comment (ymc9) [inline packages/schema/src/language-server/validator/schema-validator.ts:26]: Is this check moved elsewhere?

File: packages/schema/src/language-server/zmodel-scope.ts [hunk 3]
```diff
@@ -42,3 +52,32 @@ export class ZModelScopeComputation extends DefaultScopeComputation {
...
+        super(services);
+    }
+
+    protected override getGlobalScope(referenceType: string, context: ReferenceInfo): Scope {
+        const model = getContainerOfType(context.container, isModel);
+        if (!model) {
+            return EMPTY_SCOPE;
+        }
+
+        model.declarations.filter((x) => isPlugin(x));
```
Comment (ymc9) [inline packages/schema/src/language-server/zmodel-scope.ts:67]: Is this not needed?

File: packages/schema/src/language-server/validator/schema-validator.ts (hunk 2 above)
Comment (jiashengguo) [reply packages/schema/src/language-server/validator/schema-validator.ts:26]: Added the check in the `LoadDocument` of CLI after imported modules merged.

**Pull request merged**